readthedocs/core/unresolver.py

```diff
@@ -260,12 +311,7 @@ def _match_subproject(self, parent_project, path, external_version_slug=None):
                 check_subprojects=False,
                 external_version_slug=external_version_slug,
             )
-            # If we got a valid response, return that,
-            # otherwise return the current subproject
-            # as the current project without a valid version or path.
-            if response:
-                return response
-            return subproject, None, None
+            return response
         return None
```

Comment:
Same here. We are returning `None` in two cases, but we are not saying that in the docstring. I'd say that we should standardize these "invalid scenarios" and raise a known exception as well.

Reply:
Or, if we really require the `None` as a valid value for the workflow, we should clearly document this in the docstrings and explain how this special value is used.